backend/firebaseFunctions.js

```diff
@@ -70,28 +70,76 @@ export async function createCircle(data) {
 	data.admin[`${UID}`] = true;
 	data.owner = UID;
 
+	let circles = Object.keys((await readData(`prayer_circle/circles`)) || {});
+	const allAdminCodes = new Array(circles.length);
+	const allPublicCodes = new Array(circles.length);
+	for (let i = 0; i < circles.length; i++) {
+		let circle = circles[i];
+		let circleData =
+			(await readData(`prayer_circle/circles/${circle}`)) || {};
+		let circleStruct = {
+			adminCode: circleData.adminCode,
+			publicCode: circleData.publicCode
+		};
+		allAdminCodes[i] = circleStruct.adminCode;
+		allPublicCodes[i] = circleStruct.publicCode;
+	}
+
+	for (let i = 0; i < 2; i++) {
+		let unusedCode = false;
+		while (!unusedCode) {
+			unusedCode = true;
+			if (i === 0) {
+				data.adminCode = Math.floor(
+					Math.random() * 90000000 + 10000000
+				);
+				for (let j = 0; j < allAdminCodes.length; j++) {
+					if (allAdminCodes[j] === data.adminCode) {
+						unusedCode = false;
+					}
+				}
+			} else {
+				data.publicCode = Math.floor(
+					Math.random() * 90000000 + 10000000
+				);
+				if (data.publicCode === data.adminCode) {
+					unusedCode = false;
+				} else {
+					for (let j = 0; j < allPublicCodes.length; j++) {
+						if (allPublicCodes[j] === data.publicCode) {
+							unusedCode = false;
+						}
+					}
+				}
+			}
+		}
+	}
+
+	data.members[`${UID}`] = true;
+	data.admin[`${UID}`] = true;
+	data.owner = UID;
 	let circlePermissions = {
 		admin: true,
 		read: true,
 		write: true,
 		owner: true
 	};
 
-	/* writeData(`prayer_circle/circles/${circleId}`, data, true);
+	writeData(`prayer_circle/circles/${circleId}`, data, true);
 	writeData(
 		`prayer_circle/users/${UID}/private/circles/${circleId}/permissions`,
 		circlePermissions,
 		true
-	); */
+	);
 }
 
-export async function registerUser(username, email, password, data) {
+export async function registerUser(email, password, data) {
```

Comment:
Looks like you merged main in and this conflict was not resolved properly. Please add username back here. 
```suggestion
export async function registerUser(username, email, password, data) {
```